### PR TITLE
ref(py3): Ensure JWT tokens are used as bytes in headers

### LINKED ISSUE #712: Error when trying to access the github integration settings
An error happens when trying to open the github integration settings and it wont work. I just noticed this today and it was working before, so I don't know when it started happening.

Here are more details regarding the internal error: /settings/0soft/integrations/github/1/

It's worth noting that I'm using `SENTRY_PYTHON3=1` for that deployment, which might point to a python3 problem (but may be related to something else).

I noticed that I was receiving since last week some emails saying that "sentry was unable to fetch commits" from some deploys that I was making. That might be also related to the issue.

### PR DESCRIPTION
**Pull request by BYK**
The API reference says the return value of `jwt.encode()` is `bytes`: https://pyjwt.readthedocs.io/en/latest/api.html#jwt.encode

Should resolve getsentry/onpremise#712

### CODE REVIEW

**Comment by mitsuhiko** [comment]
Aren't headers latin1 unicode on Python3?

**Comment by BYK** [comment]
>Aren't headers latin1 unicode on Python3?

@mitsuhiko I guess they are? As you can see I have no idea about what I'm doing here right now 😁

File: .vscode/settings.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
   "python.linting.pylintEnabled": false,
   "python.linting.flake8Enabled": true,
   // https://github.com/DonJayamanne/pythonVSCode/issues/992
-  "python.pythonPath": "${workspaceFolder}/.venv/bin/python",
+  "python.pythonPath": "C:\\Users\\byk\\AppData\\Local\\Programs\\Python\\Python38\\python.exe",
```
Comment (wedamija) [inline .vscode/settings.json:59]: lol

**Review by wedamija — APPROVED** [review]

**Comment by BYK** [comment]
Just got verification that this solves the issue. Merging away!

**Pull request merged**